core/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java

```diff
@@ -219,6 +256,26 @@ protected void parseCreateField(ParseContext context, List<IndexableField> field
             return;
         }
 
+        if (normalizer != null) {
+            try (final TokenStream ts = normalizer.tokenStream(name(), value)) {
+                final CharTermAttribute termAtt = ts.addAttribute(CharTermAttribute.class);
+                ts.reset();
+                if (ts.incrementToken() == false) {
+                  throw new IllegalStateException("The normalization token stream is "
+                      + "expected to produce exactly 1 token, but got 0 for analyzer "
+                      + normalizer + " and input \"" + value + "\"");
+                }
+                final String newValue = termAtt.toString();
+                if (ts.incrementToken()) {
+                  throw new IllegalStateException("The normalization token stream is "
+                      + "expected to produce exactly 1 token, but got 2+ for analyzer "
+                      + normalizer + " and input \"" + value + "\"");
```

Comment:
Is it something that can actually happen ? The char and token filters are checked during the creation of the normalizer so it should never happen ?